docs/05-go-client/21-worker-auto-scaling.md

```diff
@@ -0,0 +1,144 @@
+---
+layout: default
+title: Worker auto scaling
+permalink: /docs/go-client/worker-auto-scaling
+---
+
+## Overview
+
+### What AutoScaler does
+
+Cadence Worker AutoScaler automatically adjusts your worker configuration to optimize resource utilization and prevent common scaling issues. Instead of manually tuning poller counts, AutoScaler monitors real-time metrics from your workers and the Cadence service to make intelligent scaling decisions.
+
+The AutoScaler addresses these critical production problems:
+- **Insufficient throughput capacity**: Automatically scales up pollers when task load increases, ensuring workflows are processed without delays
+- **Better resource utilization**: By adjusting poller counts based on actual task demand rather than static configuration, workers utilize their allocated CPU resources more efficiently, preventing unnecessary downscaling by compute autoscalers (e.g. AWS EC2 Auto Scaling)
+- **Manual configuration complexity**: Eliminates the need for service owners to understand and tune complex worker parameters
+
+### Key benefits
+
+- **Zero-configuration scaling**: Works out of the box with sensible defaults
+- **Improved resource efficiency**: Automatically scales up when needed, scales down when idle
+- **Reduced operational overhead**: No more manual tuning of poller counts and execution limits
+- **Production reliability**: Prevents scaling-related incidents and workflow processing delays
+
+### How to get started
+To get started, just add the following to your worker options.
+```go
+worker.Options{
+    ...
+    AutoScalerOptions: worker.AutoScalerOptions{
```

Comment:
@shijiesheng do you think we should also mention the top level options discarded after setting this to `true`?

Reply:
right. Here is an example
<img width="1035" height="211" alt="Screenshot 2025-08-04 at 1 21 32 PM" src="https://github.com/user-attachments/assets/58edeb37-da14-497c-8039-4a81d5034844" />